### PR TITLE
Remove a few obsolete dependencies

### PR DESCRIPTION
**Pull request by ockham**
- Bump @automattic/calypso-build to ^1.0.0-beta.1, and update `webpack.config.js` accordingly
- Remove a number of obsolete dependencies

Verify that blocks still build.

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -47,12 +47,15 @@ const editorScript = [
...
 
-const webpackConfig = getBaseWebpackConfig( null, {
-	entry: {
-		editor: editorScript,
-		...viewBlocksScripts,
-	},
-	'output-path': path.join( __dirname, 'dist' ),
-} );
+const webpackConfig = getBaseWebpackConfig(
+	{ WP: true },
```
Comment (ockham) [inline webpack.config.js]: See https://github.com/Automattic/wp-calypso/pull/32042/files#diff-b99af7f62e015cd796a25e0d65088f6aR110

**Comment by ockham** [comment]
Hmm, so that's npm trying to rebuild `node-sass`. The I've run into that error before -- typically when switching node versions in between builds (e.g. if I forgot to run `nvm use`). Typically, removing `node_modules/` (followed by a re-install) fixed that for me.

Can you give that a try (to see if this specific error goes away)? (I just tried locally, and am getting a different error -- `mini-css-extract-plugin-with-rtl` is missing `node-sass` as a dependency. Which is weird, I'll look into it :roll_eyes: )

**Comment by jeffersonrabb** [comment]
> Typically, removing node_modules/ (followed by a re-install) fixed that for me.

Hmmm, I'm still seeing the same error after blowing away `node_modules` and reinstalling. FWIW I'm running Node v10.13.0.

**Comment by ockham** [comment]
Should be ready for another try now -- works in my local testing now. Using node 10.15.3 here.

```
rm -rf node_modules/
npm run clean && npm install
npm run build
```

**Comment by jeffersonrabb** [comment]
> Should be ready for another try now -- works in my local testing now. Using node 10.15.3 here.

Hmmmm still having same issue. Updated node to 10.5.3, destroyed `node_modules` then `npm install` and `npm run build` and I still get the following:

```
MacBook-Pro:newspack-blocks jeffrabb$ npm run build

> newspack-blocks@0.1.0 build /Users/jeffrabb/Repos/automattic/newspack/VVV/www/wordpress-trunk/public_html/src/wp-content/plugins/newspack-blocks
> webpack

(node:10562) DeprecationWarning: Tapable.plugin is deprecated. Use new API on `.hooks` instead
Hash: 9a7db8f5d79abcd3c45d
Version: webpack 4.29.6
Time: 742ms
Built at: 04/09/2019 10:28:39 PM
                             Asset      Size                    Chunks             Chunk Names
                  editor.deps.json  64 bytes                    editor  [emitted]  editor
                         editor.js  78.4 KiB                    editor  [emitted]  editor
example2-attributes/view.deps.json   2 bytes  example2-attributes/view  [emitted]  example2-attributes/view
       example2-attributes/view.js  8.83 KiB  example2-attributes/view  [emitted]  example2-attributes/view
       example3-ssr/view.deps.json   2 bytes         example3-ssr/view  [emitted]  example3-ssr/view
              example3-ssr/view.js   8.7 KiB         example3-ssr/view  [emitted]  example3-ssr/view
Entrypoint editor = editor.js editor.deps.json
Entrypoint example2-attributes/view = example2-attributes/view.js example2-attributes/view.deps.json
Entrypoint example3-ssr/view = example3-ssr/view.js example3-ssr/view.deps.json
[0] multi ./src/setup/editor ./src/blocks/example1-simple/editor.js ./src/blocks/example2-attributes/editor.js ./src/blocks/example3-ssr/editor.js ./src/blocks/test/editor.js 76 bytes {editor} [built]
[1] multi ./src/setup/view ./src/blocks/example2-attributes/view.js 40 bytes {example2-attributes/view} [built]
[2] multi ./src/setup/view ./src/blocks/example3-ssr/view.js 40 bytes {example3-ssr/view} [built]
[./src/blocks/example1-simple/editor.js] 185 bytes {editor} [built]
[./src/blocks/example1-simple/index.js] 1.19 KiB {editor} [built]
[./src/blocks/example2-attributes/editor.js] 185 bytes {editor} [built]
[./src/blocks/example2-attributes/index.js] 1.35 KiB {editor} [built]
[./src/blocks/example2-attributes/style.scss] 1.85 KiB {example2-attributes/view} [built] [failed] [1 error]
[./src/blocks/example2-attributes/view.js] 22 bytes {example2-attributes/view} [built]
[./src/blocks/example3-ssr/editor.js] 185 bytes {editor} [built]
[./src/blocks/example3-ssr/index.js] 1.18 KiB {editor} [built]
[./src/blocks/example3-ssr/view.js] 22 bytes {example3-ssr/view} [built]
[./src/blocks/test/editor.js] 185 bytes {editor} [built]
[./src/setup/editor.js] 77 bytes {editor} [built]
[./src/setup/view.js] 56 bytes {example2-attributes/view} {example3-ssr/view} [built]
    + 30 hidden modules

ERROR in ./src/blocks/example2-attributes/editor.scss
Module build failed (from ./node_modules/@automattic/calypso-build/node_modules/mini-css-extract-plugin-with-rtl/dist/loader.js):
ModuleBuildError: Module build failed (from ./node_modules/sass-loader/lib/loader.js):

.wp-block-newspack-blocks-example2-attributes {
^
      File to import not found or unreadable: ~@automattic/calypso-color-schemes/src/shared/colors.
      in /Users/jeffrabb/Repos/automattic/newspack/VVV/www/wordpress-trunk/public_html/src/wp-content/plugins/newspack-blocks/src/blocks/example2-attributes/editor.scss (line 1, column 1)
    at runLoaders (/Users/jeffrabb/Repos/automattic/newspack/VVV/www/wordpress-trunk/public_html/src/wp-content/plugins/newspack-blocks/node_modules/webpack/lib/NormalModule.js:301:20)
    at /Users/jeffrabb/Repos/automattic/newspack/VVV/www/wordpress-trunk/public_html/src/wp-content/plugins/newspack-blocks/node_modules/loader-runner/lib/LoaderRunner.js:367:11
    at /Users/jeffrabb/Repos/automattic/newspack/VVV/www/wordpress-trunk/public_html/src/wp-content/plugins/newspack-blocks/node_modules/loader-runner/lib/LoaderRunner.js:233:18
    at context.callback (/Users/jeffrabb/Repos/automattic/newspack/VVV/www/wordpress-trunk/public_html/src/wp-content/plugins/newspack-blocks/node_modules/loader-runner/lib/LoaderRunner.js:111:13)
    at Object.render [as callback] (/Users/jeffrabb/Repos/automattic/newspack/VVV/www/wordpress-trunk/public_html/src/wp-content/plugins/newspack-blocks/node_modules/sass-loader/lib/loader.js:52:13)
    at Object.done [as callback] (/Users/jeffrabb/Repos/automattic/newspack/VVV/www/wordpress-trunk/public_html/src/wp-content/plugins/newspack-blocks/node_modules/neo-async/async.js:8077:18)
    at options.error (/Users/jeffrabb/Repos/automattic/newspack/VVV/www/wordpress-trunk/public_html/src/wp-content/plugins/newspack-blocks/node_modules/node-sass/lib/index.js:294:32)
 @ ./src/blocks/example2-attributes/edit.js 16:0-23
 @ ./src/blocks/example2-attributes/index.js
 @ ./src/blocks/example2-attributes/editor.js
 @ multi ./src/setup/editor ./src/blocks/example1-simple/editor.js ./src/blocks/example2-attributes/editor.js ./src/blocks/example3-ssr/editor.js ./src/blocks/test/editor.js

ERROR in ./src/blocks/example2-attributes/style.scss
Module build failed (from ./node_modules/@automattic/calypso-build/node_modules/mini-css-extract-plugin-with-rtl/dist/loader.js):
ModuleBuildError: Module build failed (from ./node_modules/sass-loader/lib/loader.js):

.wp-block-newspack-blocks-example2-attributes {
^
      File to import not found or unreadable: ~@automattic/calypso-color-schemes/src/shared/colors.
      in /Users/jeffrabb/Repos/automattic/newspack/VVV/www/wordpress-trunk/public_html/src/wp-content/plugins/newspack-blocks/src/blocks/example2-attributes/style.scss (line 1, column 1)
    at runLoaders (/Users/jeffrabb/Repos/automattic/newspack/VVV/www/wordpress-trunk/public_html/src/wp-content/plugins/newspack-blocks/node_modules/webpack/lib/NormalModule.js:301:20)
    at /Users/jeffrabb/Repos/automattic/newspack/VVV/www/wordpress-trunk/public_html/src/wp-content/plugins/newspack-blocks/node_modules/loader-runner/lib/LoaderRunner.js:367:11
    at /Users/jeffrabb/Repos/automattic/newspack/VVV/www/wordpress-trunk/public_html/src/wp-content/plugins/newspack-blocks/node_modules/loader-runner/lib/LoaderRunner.js:233:18
    at context.callback (/Users/jeffrabb/Repos/automattic/newspack/VVV/www/wordpress-trunk/public_html/src/wp-content/plugins/newspack-blocks/node_modules/loader-runner/lib/LoaderRunner.js:111:13)
    at Object.render [as callback] (/Users/jeffrabb/Repos/automattic/newspack/VVV/www/wordpress-trunk/public_html/src/wp-content/plugins/newspack-blocks/node_modules/sass-loader/lib/loader.js:52:13)
    at Object.done [as callback] (/Users/jeffrabb/Repos/automattic/newspack/VVV/www/wordpress-trunk/public_html/src/wp-content/plugins/newspack-blocks/node_modules/neo-async/async.js:8077:18)
    at options.error (/Users/jeffrabb/Repos/automattic/newspack/VVV/www/wordpress-trunk/public_html/src/wp-content/plugins/newspack-blocks/node_modules/node-sass/lib/index.js:294:32)
 @ ./src/blocks/example2-attributes/view.js 1:0-22
 @ multi ./src/setup/view ./src/blocks/example2-attributes/view.js

ERROR in ./src/blocks/example3-ssr/style.scss
Module build failed (from ./node_modules/@automattic/calypso-build/node_modules/mini-css-extract-plugin-with-rtl/dist/loader.js):
ModuleBuildError: Module build failed (from ./node_modules/sass-loader/lib/loader.js):

.wp-block-newspack-blocks-example3-ssr {
^
      File to import not found or unreadable: ~@automattic/calypso-color-schemes/src/shared/colors.
      in /Users/jeffrabb/Repos/automattic/newspack/VVV/www/wordpress-trunk/public_html/src/wp-content/plugins/newspack-blocks/src/blocks/example3-ssr/style.scss (line 1, column 1)
    at runLoaders (/Users/jeffrabb/Repos/automattic/newspack/VVV/www/wordpress-trunk/public_html/src/wp-content/plugins/newspack-blocks/node_modules/webpack/lib/NormalModule.js:301:20)
    at /Users/jeffrabb/Repos/automattic/newspack/VVV/www/wordpress-trunk/public_html/src/wp-content/plugins/newspack-blocks/node_modules/loader-runner/lib/LoaderRunner.js:367:11
    at /Users/jeffrabb/Repos/automattic/newspack/VVV/www/wordpress-trunk/public_html/src/wp-content/plugins/newspack-blocks/node_modules/loader-runner/lib/LoaderRunner.js:233:18
    at context.callback (/Users/jeffrabb/Repos/automattic/newspack/VVV/www/wordpress-trunk/public_html/src/wp-content/plugins/newspack-blocks/node_modules/loader-runner/lib/LoaderRunner.js:111:13)
    at Object.render [as callback] (/Users/jeffrabb/Repos/automattic/newspack/VVV/www/wordpress-trunk/public_html/src/wp-content/plugins/newspack-blocks/node_modules/sass-loader/lib/loader.js:52:13)
    at Object.done [as callback] (/Users/jeffrabb/Repos/automattic/newspack/VVV/www/wordpress-trunk/public_html/src/wp-content/plugins/newspack-blocks/node_modules/neo-async/async.js:8077:18)
    at options.error (/Users/jeffrabb/Repos/automattic/newspack/VVV/www/wordpress-trunk/public_html/src/wp-content/plugins/newspack-blocks/node_modules/node-sass/lib/index.js:294:32)
 @ ./src/blocks/example3-ssr/view.js 1:0-22
 @ multi ./src/setup/view ./src/blocks/example3-ssr/view.js
Child mini-css-extract-plugin node_modules/css-loader/dist/cjs.js??ref--5-1!node_modules/postcss-loader/src/index.js??ref--5-2!node_modules/sass-loader/lib/loader.js??ref--5-3!src/blocks/example2-attributes/editor.scss:
    Entrypoint mini-css-extract-plugin = *
    [./node_modules/css-loader/dist/cjs.js?!./node_modules/postcss-loader/src/index.js?!./node_modules/sass-loader/lib/loader.js?!./src/blocks/example2-attributes/editor.scss] ./node_modules/css-loader/dist/cjs.js??ref--5-1!./node_modules/postcss-loader/src??ref--5-2!./node_modules/sass-loader/lib/loader.js??ref--5-3!./src/blocks/example2-attributes/editor.scss 432 bytes {mini-css-extract-plugin} [built] [failed] [1 error]
    
    ERROR in ./src/blocks/example2-attributes/editor.scss (./node_modules/css-loader/dist/cjs.js??ref--5-1!./node_modules/postcss-loader/src??ref--5-2!./node_modules/sass-loader/lib/loader.js??ref--5-3!./src/blocks/example2-attributes/editor.scss)
    Module build failed (from ./node_modules/sass-loader/lib/loader.js):
    
    .wp-block-newspack-blocks-example2-attributes {
    ^
          File to import not found or unreadable: ~@automattic/calypso-color-schemes/src/shared/colors.
          in /Users/jeffrabb/Repos/automattic/newspack/VVV/www/wordpress-trunk/public_html/src/wp-content/plugins/newspack-blocks/src/blocks/example2-attributes/editor.scss (line 1, column 1)
Child mini-css-extract-plugin node_modules/css-loader/dist/cjs.js??ref--5-1!node_modules/postcss-loader/src/index.js??ref--5-2!node_modules/sass-loader/lib/loader.js??ref--5-3!src/blocks/example2-attributes/style.scss:
    Entrypoint mini-css-extract-plugin = *
    [./node_modules/css-loader/dist/cjs.js?!./node_modules/postcss-loader/src/index.js?!./node_modules/sass-loader/lib/loader.js?!./src/blocks/example2-attributes/style.scss] ./node_modules/css-loader/dist/cjs.js??ref--5-1!./node_modules/postcss-loader/src??ref--5-2!./node_modules/sass-loader/lib/loader.js??ref--5-3!./src/blocks/example2-attributes/style.scss 431 bytes {mini-css-extract-plugin} [built] [failed] [1 error]
    
    ERROR in ./src/blocks/example2-attributes/style.scss (./node_modules/css-loader/dist/cjs.js??ref--5-1!./node_modules/postcss-loader/src??ref--5-2!./node_modules/sass-loader/lib/loader.js??ref--5-3!./src/blocks/example2-attributes/style.scss)
    Module build failed (from ./node_modules/sass-loader/lib/loader.js):
    
    .wp-block-newspack-blocks-example2-attributes {
    ^
          File to import not found or unreadable: ~@automattic/calypso-color-schemes/src/shared/colors.
          in /Users/jeffrabb/Repos/automattic/newspack/VVV/www/wordpress-trunk/public_html/src/wp-content/plugins/newspack-blocks/src/blocks/example2-attributes/style.scss (line 1, column 1)
Child mini-css-extract-plugin node_modules/css-loader/dist/cjs.js??ref--5-1!node_modules/postcss-loader/src/index.js??ref--5-2!node_modules/sass-loader/lib/loader.js??ref--5-3!src/blocks/example3-ssr/style.scss:
    Entrypoint mini-css-extract-plugin = *
    [./node_modules/css-loader/dist/cjs.js?!./node_modules/postcss-loader/src/index.js?!./node_modules/sass-loader/lib/loader.js?!./src/blocks/example3-ssr/style.scss] ./node_modules/css-loader/dist/cjs.js??ref--5-1!./node_modules/postcss-loader/src??ref--5-2!./node_modules/sass-loader/lib/loader.js??ref--5-3!./src/blocks/example3-ssr/style.scss 417 bytes {mini-css-extract-plugin} [built] [failed] [1 error]
    
    ERROR in ./src/blocks/example3-ssr/style.scss (./node_modules/css-loader/dist/cjs.js??ref--5-1!./node_modules/postcss-loader/src??ref--5-2!./node_modules/sass-loader/lib/loader.js??ref--5-3!./src/blocks/example3-ssr/style.scss)
    Module build failed (from ./node_modules/sass-loader/lib/loader.js):
    
    .wp-block-newspack-blocks-example3-ssr {
    ^
          File to import not found or unreadable: ~@automattic/calypso-color-schemes/src/shared/colors.
          in /Users/jeffrabb/Repos/automattic/newspack/VVV/www/wordpress-trunk/public_html/src/wp-content/plugins/newspack-blocks/src/blocks/example3-ssr/style.scss (line 1, column 1)
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! newspack-blocks@0.1.0 build: `webpack`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the newspack-blocks@0.1.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/jeffrabb/.npm/_logs/2019-04-10T02_28_40_251Z-debug.log
MacBook-Pro:newspack-blocks jeffrabb$ 

```

**Comment by ockham** [comment]
Per a screensharing call, while things work for me, they don't for Jeff :slightly_frowning_face: There's no `calypso-color-schemes` subdir in his `node_modules/@automattic`, while there is for me.

**Comment by simison** [comment]
Worked for me. @jeffersonrabb do you have Xcode with command line tools installed?

Those are needed if you need to compile code.

```
$ node --version

v10.15.3

$ gcc --version

Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/usr/include/c++/4.2.1
Apple LLVM version 10.0.1 (clang-1001.0.46.3)
Target: x86_64-apple-darwin18.5.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

I also have `sass` globally installed if it matters:

```
$ sass --version
Ruby Sass 3.7.3
```

**Comment by jeffersonrabb** [comment]
> Worked for me (MacOS Mojave 10.14.4). @jeffersonrabb do you have Xcode with command line tools installed?

Sorry to be such a downer 😞I'm on a relatively new Mac and hadn't opened XCode in ages, so I launched, allowed updates to take place, and restarted—no luck. Here's my version info for node, gcc and sass, which looks similar to @simison's:

```$ node --version
v10.15.3
```

```$ gcc --version
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/usr/include/c++/4.2.1
Apple LLVM version 10.0.1 (clang-1001.0.46.3)
Target: x86_64-apple-darwin18.5.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin```

```$ sass --version
Sass 3.4.22 (Selective Steve)
```

**Comment by jeffersonrabb** [comment]
Rebased...

**Comment by jeffersonrabb** [comment]
Turned out the problem here was that `package-lock.json` was [not being committed](https://github.com/Automattic/newspack-blocks/commit/4299e3dc248749a42febf6289936894b144b955c). The errors cleared up if I ran `rm -rf node_modules package-lock.json` after branch switches. I  suspect other people didn't see the same errors because they checked out `update/simplify-build` _first_, rather than running `npm install` on `master` and then switching. In any case, my problems are resolved as of https://github.com/Automattic/newspack-blocks/pull/6/commits/b0293e701c7c58c84132c6b51bd58a4aacd9654f

**Comment by ockham** [comment]
Glad you found the issue, Jeff! :tada: 

Weirdly enough, `npm run build` now gives me an error,

```
Error: Cannot find module '@wordpress/browserslist-config'
```

This happens both on `master` and this branch, so it's arguably not a regression. (Just weird that we're always seeing different errors.)

That dep probably needs to be added to `calypso-build`, I'll take care of that.

**Pull request merged**